source/application/utils/file/FileDialog.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright(c) 2017- Hourglass Resurrection Team
+ * Hourglass Resurrection is licensed under GPL v2.
+ * Refer to the file COPYING.txt in the project root.
+ */
+
+#include <windows.h>
+#include <CommDlg.h>
+
+#include <map>
+#include <algorithm>
+
+#include "../File.h"
+
+namespace
+{
+    // The pipe characters are just placeholders for the null characters to ease string manipulation
+    std::map<Utils::File::FileFilter, const char*> file_type_map =
+    {
+        { Utils::File::FileFilter::AllFiles, "All Files (*.*)|*.*|" },
+        { Utils::File::FileFilter::HourglassMovie, "Windows TAS Files (*.hgr)|*.hgr|" },
+        { Utils::File::FileFilter::Executable, "Executables (*.exe)|*.exe|" },
+        { Utils::File::FileFilter::Config, "Config Files (*.cfg)|*.cfg|" },
+        { Utils::File::FileFilter::WatchList, "Watchlist (*.wch)|*.wch|" },
+        { Utils::File::FileFilter::AVI, "AVI file (*.avi)|*.avi|" }
+    };
+    
+    std::string GetFilter(const std::vector<Utils::File::FileFilter> file_types)
+    {
+        std::string result = "";
+
+        for (auto& file_type : file_types)
+        {
+            const char* filter = file_type_map[file_type];
+            result.append(filter);
+        }
+
+        std::replace(result.begin(), result.end(), '|', '\0');
+        return result;
+    }
+}
+
+std::string Utils::File::GetFileNameOpen(
```

Comment:
Oh the code duplication between Open and Save. Can you consolidate this? Perhaps with an `enum class` to pick Open or Save?